Explorer/Assets/DCL/Profiles/ProfileJsonDto.cs

```diff
@@ -0,0 +1,205 @@
+            avatar.EyesColor = eyes.color.ToColor();
+            avatar.HairColor = hair.color.ToColor();
+            avatar.SkinColor = skin.color.ToColor();
+        }
+
+        public void Reset()
+        {
+            bodyShape = default(string?);
+            wearables.Clear();
+            forceRender?.Clear();
+            emotes.Clear();
+            snapshots.face256 = default(string);
+            snapshots.body = default(string);
+            eyes.color.Reset();
+            hair.color.Reset();
+            skin.color.Reset();
+        }
+    }
+
+    [Serializable]
+    public class ProfileJsonDto : IDisposable
+    {
+        private static readonly ThreadSafeObjectPool<ProfileJsonDto> pool = new (() => new ProfileJsonDto());
+
+        public bool hasClaimedName;
+        public string description;
+        public int tutorialStep;
+        public string name;
+        public string userId;
+        public string email;
+        public string ethAddress;
+        public int version;
+        public AvatarJsonDto? avatar;
+        public List<string>? blocked;
+        public List<string>? interests;
+        public string unclaimedName;
+        public bool hasConnectedWeb3;
+
+        public static ProfileJsonDto Create()
+        {
+            ProfileJsonDto profile = pool.Get();
+            profile.Reset();
+            return profile;
+        }
+
+        public void Dispose()
+        {
+            pool.Release(this);
+        }
+
+        public void CopyTo(Profile profile)
+        {
+            profile.UserId = userId;
+            profile.Name = name;
+            profile.UnclaimedName = unclaimedName;
+            profile.HasClaimedName = hasClaimedName;
+            profile.Description = description;
+            profile.TutorialStep = tutorialStep;
+            profile.Email = email;
+            profile.Version = version;
+            profile.Avatar ??= new Avatar();
+            avatar?.CopyTo(profile.Avatar);
+            profile.Blocked = blocked != null ? new HashSet<string>(blocked) : new HashSet<string>();
+            profile.Interests = interests != null ? new List<string>(interests) : new List<string>();
+        }
+
+        private void Reset()
+        {
+            hasClaimedName = default(bool);
+            description = default(string);
+            tutorialStep = default(int);
+            name = default(string);
+            userId = default(string);
+            email = default(string);
+            ethAddress = default(string);
+            version = default(int);
+            avatar?.Reset();
+            blocked?.Clear();
+            interests?.Clear();
+            unclaimedName = default(string);
+            hasConnectedWeb3 = default(bool);
+        }
+    }
+
+    [Serializable]
+    public class GetProfileJsonRootDto : IDisposable
+    {
+        private static readonly ThreadSafeObjectPool<GetProfileJsonRootDto> pool = new (() => new GetProfileJsonRootDto());
```

Comment:
Should the `GetProfileJsonRootDto` be kept as a simple data transfer object and be decoupled from the object pooling logic?

Reply:
As for me, it's fine to keep pools where they belong to. Otherwise we need extra utility classes that lead to fragmentation. With this design Factory methods and pools are defined exactly in the visible scope